### ISSUE TITLE
Feature: Provide named exports for public APIs

### ISSUE THREAD

**Issue by iamzapata** [facebook/lexical#2267]
### Context

I just started using this library recently I've had a few issues, but I'm really excited about it. One thing I find annoying is that some examples are inconsistent. Let me explain. 

If you go to the [@lexical/react README](https://github.com/facebook/lexical/tree/main/packages/lexical-react) and copy/paste the sample code, it won't work with the latest version (0.2.9 as of May 2022). 

If you open the [codesandbox](https://codesandbox.io/s/lexical-plain-text-example-g932e) it does work. 

What's the difference? 

The difference is that in the `README`, the example is importing **named** modules: 

```javascript
import {LexicalComposer} from '@lexical/react/LexicalComposer';
import {LexicalPlainTextPlugin} from '@lexical/react/LexicalPlainTextPlugin';
import {LexicalContentEditable} from '@lexical/react/LexicalContentEditable';
```

 and in the `codesandbox`, the code is **importing** default modules:
 
 ```javascript
 import LexicalComposer from "@lexical/react/LexicalComposer";
import PlainTextPlugin from "@lexical/react/LexicalPlainTextPlugin";
import ContentEditable from "@lexical/react/LexicalContentEditable";
 ```
 
 `@lexical/react@0.2.9` uses default exports and **not** named exports.

### Ideal Solution

I think being able to import modules in a single **from** would be ideal, and should also apply to the other lexical packages.

```javascript
 import { LexicalComposer, PlainTextPlugin, ContentEditable } from "@lexical/react";
```

**Comment by trueadm** [comment]
I agree that importing from a single module seems nicer, but there’s a few things wrong with that approach. Firstly, Lexical works with any framework, so having exports for React specific things doesn’t make too much sense. Furthermore, those who cannot apply treeshaking mean you pay a huge cost for code you might not use.

**Comment by iamzapata** [comment]
@trueadm thank you for your answer 👍
